design_matrix_creation_module.py

```diff
@@ -0,0 +1,19 @@
+import numpy as np
+import pandas as pd
+
+
+def get_frame_at_time(time, stim_table):
```

Comment:
There are a couple of oddities about this function. One is that it selects the output column by integer index rather than name, which will fail if you run into a stim table with differently-ordered or inserted columns. The other is that you are looking at start times and not end times, so if you supply a time after the bounds of the experiment, this function will happily report the last value from the dataframe (rather than throwing an error or otherwise notifying the caller).

Maybe something like this (sans error handling):
```
def get_frame_at_time(time, stim_table):
    return stim_table.loc[(stim_table['start'] <= time) & (stim_table['end'] > time), 'frame'].values[0]
```